apps/teamsjs-diagnostic-app/src/apis/AppInstallDialogApi.tsx

```diff
@@ -1,8 +1,72 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDrag } from 'react-dnd';
+import { ApiComponent } from '../components/sample/ApiComponents';
+
+interface AppInstallDialogAPIsProps {
+  apiComponent: ApiComponent;
+  onDropToScenarioBox: (apiComponent: ApiComponent, func: string, input: string) => void;
+}
+
+const AppInstallDialogAPIs: React.FC<AppInstallDialogAPIsProps> = ({ apiComponent, onDropToScenarioBox }) => {
+  const [selectedFunction, setSelectedFunction] = useState<string>('');
+  const [inputValue, setInputValue] = useState<string>('');
+
+  const handleFunctionChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedFunc = event.target.value;
+    setSelectedFunction(selectedFunc);
```

Comment:
Something to be aware of: each time you call a `set*` function in React (`setSelectedFunction` and `setInputValue` for example), each function call will re-render the react component. That means that your `handleFunctionChange` function will be causing two renders, one where selectedFunc is set to the new value, and a second one where you have selectedFunc and inputvalue set.

That might not be a problem (it depends on how you've put everything together), but it's worth being aware of. If it is causing bugs (or performance problems) you could put your two pieces of data together into a single interface, something like:

```typescript
const [appInstallProps, setAppInstallProps] = useState<AppInstallProps | null>(null);

interface AppInstallProps {
  selectedFunction: string;
  inputValue: string;
}
```

Reply:
Right now, it's not causing any bugs, but I will keep an eye out for this! Thank you!